Wrapping/Generators/Python/itk/support/template_class.py

```diff
@@ -718,6 +719,20 @@ def ttype_for_input_type(keys_l, input_type_l):
             # try to find a type suitable for the input provided
             input_type = output(cur).__class__
             keys = ttype_for_input_type(keys, input_type)
+        else:
+            inst = self.values()[0].New()
+            if hasattr(inst, "GetRequiredInputNames"):
+                required_input_names = inst.GetRequiredInputNames()
+                if len(required_input_names) > 0:
+                    primary_input_name = required_input_names[0]
+                    kwargs_camel = {snake_to_camel_case(k): v for k,v in kwargs.items()}
+                    if primary_input_name in kwargs_camel.keys():
+                        input_type = output(kwargs_camel[primary_input_name]).__class__
+                        keys = ttype_for_input_type(keys, input_type)
+                        if not hasattr(inst, f"Set{primary_input_name}"):
+                            arg_0 = kwargs_camel.pop(primary_input_name)
+                            kwargs = kwargs_camel
+                            args = (arg_0,) + args
```

Comment:
@thewtex @hjmjohnson @dzenanz @blowekamp I'm very glad to see this pull request is merged, thanks! Especially thanks to Matt, of course! Sorry I wasn't able to finalize my review. I really find the added code hard to understand (especially the code here in "template_class.py"), I'm afraid it's beyond my Python skills! 🤷 

Regarding the tests, I'm glad that they do now include a comparison between the output from a keyword argument and the output from a positional argument. It took me a lot of time to get the Tests/extras.py test up and running, especially because it does so many other things than just test this specific feature. For the future, I hope that we can do more structured unit testing, using either Python `unittest` or `pytest`, having small test functions that can test one specific feature independently and efficiently.

Reply:
With the in-progress Pixi work, it will be easier to make `pytest` available for testing, which can greatly improve our test driver options.

Reply:
`pytest` would be nice, but `unittest` is already available right now, for ITK development. `unittest` already very much helps to make unit tests more well structured and easier to read, in my opinion. It's already being used in [Modules\Bridge\NumPy\wrapping\test](https://github.com/InsightSoftwareConsortium/ITK/tree/53750566cc8f0265ff189390939c23000391555d/Modules/Bridge/NumPy/wrapping/test)

Reply:
+1 for pytest use.